services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -196,25 +197,25 @@ public Response updateUser(final UserRepresentation rep) {
         }
     }
 
-    public static Response validateUserProfile(UserModel user, UserRepresentation rep, KeycloakSession session) {
-        UserProfileValidationResult result = forUserResource(user, rep, session).validate();
-        if (!result.getErrors().isEmpty()) {
-            for (AttributeValidationResult attrValidation : result.getErrors()) {
-                StringBuilder s = new StringBuilder("Failed to update attribute " + attrValidation.getField() + ": ");
-                for (ValidationResult valResult : attrValidation.getFailedValidations()) {
-                    s.append(valResult.getErrorType() + ", ");
-                }
+    public static Response validateUserProfile(UserProfile profile) {
+        try {
+            profile.validate();
+        } catch (ValidationException pve) {
+            for (ValidationException.Error error : pve.getErrors()) {
+                StringBuilder s = new StringBuilder("Failed to update attribute " + error.getAttribute() + ": ");
+
+                s.append(error.getMessage()).append(", ");
+
                 logger.warn(s);
             }
             return ErrorResponse.error("Could not update user! See server log for more details", Response.Status.BAD_REQUEST);
```

Comment:
It would be nice to get the actual validation errors instead of this generic error message - in our case the users of the REST API don't have access to the logs, so this message won't really help them. If the response should be kept simple, it would be nice to get at least the first validation error (as it is done in Account REST API).

Reply:
@danielFesenmeyer I agree. However, we are not changing here legacy behavior but re-working the SPI. Maybe we can track this in a separate JIRA?

Reply:
@pedroigor: I've created this ticket here as a subtask of the user profile: https://issues.redhat.com/browse/KEYCLOAK-17946. I hope that's the right place - otherwise feel free to move it.